### PR TITLE
RDSC-1493-ingest-quickstrat-prerequisites-redis-enterprise-cluster-version-is-not-correct

### PR DESCRIPTION
**Pull request by yaronp68**

### CODE REVIEW

File: config.toml [hunk 1]
```diff
@@ -48,11 +48,11 @@ windows_dl_link = "https://downloads.redisinsight.redislabs.com/latest/redisinsi
 docker_command = "docker run -v redisinsight:/db -p 8001:8001 redislabs/redisinsight:latest"
 docker_db_volume_permission = "chown -R 1001 redisinsight"
 
-rdi_redis_gears_min_semantic_version = "1.2.5"
-rdi_redis_gears_min_version = "10205"
+rdi_redis_gears_min_semantic_version = "1.2.6"
+rdi_redis_gears_min_version = "10206"
 rdi_redis_gears_current_semantic_version = "1.2.6"
 rdi_redis_gears_current_version = "10206"
```
Comment (spicy-sauce) [inline config.toml:54]: do we really need the 4 of them? Seems like we just need the `min_version` and `min_semantic_version` and update accordingly. No real need to say that we support 1.2.5 but the current one is 1.2.6, let's keep it simple.

File: content/rdi/architecture.md [hunk 2]
```diff
@@ -9,59 +9,89 @@ headerRange: "[2]"
 aliases: 
 ---
 
-Redis Data Integration is a product for data ingestion from existing systems of record (mainly databases) into Redis Enterprise.
-You can think of Redis Data Integration as a streaming ELT process:
+Redis Data Integration (RDI) is a product that helps Redis Enterprise users ingest and export data in near real time. RDI provides the following:
```
Comment (spicy-sauce) [inline content/rdi/architecture.md:12]: I'd omit the product:
Redis Data Integration (RDI) helps...

Comment (spicy-sauce) [inline content/rdi/architecture.md:12]: Consider having the "RDI provides the following" in a new line (probably with another empty line) to have it in another section.

File: content/rdi/architecture.md [hunk 3]
```diff
@@ -9,59 +9,89 @@ headerRange: "[2]"
...
 
-- Data is **E**xtracted from the source database using [Debezium](https://debezium.io/) connectors.
+- End to end solution - No need for additional tools and integrations
+- Capture Data Change (CDC) included
+- Covers most popular databases as sources and targets
+- Declarative data mapping and transformations - no custom code needed
+- Data delivery guaranteed (at least once)
+- Zero downtime day 2 operations (reconfigure & upgrade)
```
Comment (spicy-sauce) [inline content/rdi/architecture.md:19]: ChatGPT suggestion, see if some of it is better:
RDI offers the following features:
- End-to-end solution: No additional tools or integrations required
- Includes Change Data Capture (CDC)
- Supports a wide range of popular databases as both sources and targets
- Provides declarative data mapping and transformations, eliminating the need for custom code
- Ensures data delivery with at least once guarantee
- Enables zero-downtime day 2 operations for reconfiguration and upgrades

File: content/rdi/architecture.md [hunk 4]
```diff
@@ -9,59 +9,89 @@ headerRange: "[2]"
...
+
+Redis Data Integration currently supports two use cases:
```
Comment (spicy-sauce) [inline content/rdi/architecture.md:21]: Use RDI instead as you did before.

File: content/rdi/architecture.md [hunk 5]
```diff
@@ -9,59 +9,89 @@ headerRange: "[2]"
...
+
+- Ingest (cache prefetch).
+- Write Behind.
```
Comment (spicy-sauce) [inline content/rdi/architecture.md:24]: remove period.

File: content/rdi/architecture.md [hunk 6]
```diff
@@ -9,59 +9,89 @@ headerRange: "[2]"
...
+#### Mode mapping
+
+RDI supports conversion of a database record into the following Redis types:
+
+- Hash
+- Set
+- Sorted Set
+- Stream
+- JSON
+- String
```
Comment (spicy-sauce) [inline content/rdi/architecture.md:52]: I'd make each one of them a link to the proper Redis documentation page.

File: content/rdi/architecture.md [hunk 7]
```diff
@@ -9,59 +9,89 @@ headerRange: "[2]"
...
 
 ![Detailed Redis Data Integration architecture](/images/rdi/redis-di.png)
 
-## Feeders
+#### Feeders
 
-Currently the only source for Redis Data Integration is the [Debezium Redis Sink Connector](https://debezium.io/documentation/reference/stable/operations/debezium-server.html#_redis_stream).
-Other sources and SDK will be added in later versions.
+Debezium Server has a Redis sink that writes data and state to RDI streams.
+In addition RDI works with [Arcion](arcion.io), a commercial CDC that provides additional set of sources.
```
Comment (spicy-sauce) [inline content/rdi/architecture.md:69]: Consider adding an empty line above or they won't be separated.

File: content/rdi/architecture.md [hunk 8]
```diff
@@ -97,3 +127,28 @@ Debezium Server should run in warm High Availability topology in one of the foll
 - Running on 2 VMs in an active/passive setup. The failover orchestration can be done by tools such as [Pacemaker](https://clusterlabs.org/pacemaker/doc/).
 - Running as a Kubernetes pod where Kubernetes watches the pod liveliness and readiness and acts to recover nonfunctional pod
   It is important to note that Redis Data Integration does not keep any state on Debezium. All state is kept in Redis Data Integration.
+
+## Write behind functionality and architecture
+
+RDI write behind allows integration between Redis Enterprise (as the source of changes to data) to downstream databases or datastores.
```
Comment (spicy-sauce) [inline content/rdi/architecture.md:133]: you wrote Write Behind before, be consistent throughout the documentation.

File: content/rdi/architecture.md [hunk 9]
```diff
@@ -97,3 +127,28 @@ Debezium Server should run in warm High Availability topology in one of the foll
...
+RDI captures the changes to a selected set of key-patterns in a Redis keyspace and asynchronously writes them in small batches to the downstream database, so the application doesn't need to take care of remodeling the data or managing the connection with the downstream database.
+
+RDI write behind can normalize a key in Redis to several records in one or more tables at the target.
```
Comment (spicy-sauce) [inline content/rdi/architecture.md:136]: add an empty line after this or they'll be shown together.

File: content/rdi/architecture.md [hunk 10]
```diff
@@ -97,3 +127,28 @@ Debezium Server should run in warm High Availability topology in one of the foll
...
+To learn more about write behind declarative jobs and renormalization read the [write behind quick start guide]({{<relref "/rdi/quickstart/write-behind-guide.md">}})
```
Comment (spicy-sauce) [inline content/rdi/architecture.md:137]: Add period.

File: content/rdi/architecture.md [hunk 11]
```diff
@@ -97,3 +127,28 @@ Debezium Server should run in warm High Availability topology in one of the foll
...
+
+### Write behind topology
+
+RDI CLI and engine come in one edition that can run both ingest and write behind. However, the topology for write behind is different.
```
Comment (spicy-sauce) [inline content/rdi/architecture.md:141]: Ingest? Write Behind?

File: content/rdi/architecture.md [hunk 12]
```diff
@@ -97,3 +127,28 @@ Debezium Server should run in warm High Availability topology in one of the foll
...
+RDI engine is installed on a Redis database containing the application data and not on a separate staging Redis database. RDI streams and control plain adds a small footprint of few hundreds of MBs to the Redis database. In the write behind topology, RDI process in parallel on each shard and establishes a single connection from each shard to the downstream database.
```
Comment (spicy-sauce) [inline content/rdi/architecture.md:142]: add an empty line above.

File: content/rdi/architecture.md (hunk 11 above)
Comment (spicy-sauce) [inline content/rdi/architecture.md:141]: Engine?

File: content/rdi/architecture.md [hunk 13]
```diff
@@ -97,3 +127,28 @@ Debezium Server should run in warm High Availability topology in one of the foll
...
+
+### model translation
```
Comment (spicy-sauce) [inline content/rdi/architecture.md:144]: model --> Model

File: content/rdi/architecture.md [hunk 14]
```diff
@@ -97,3 +127,28 @@ Debezium Server should run in warm High Availability topology in one of the foll
...
+
+RDI write behind can track changes to the following Redis types:
```
Comment (spicy-sauce) [inline content/rdi/architecture.md:146]: RDI Write Behind?

File: content/rdi/architecture.md [hunk 15]
```diff
@@ -97,3 +127,28 @@ Debezium Server should run in warm High Availability topology in one of the foll
...
+
+- Hash
+- JSON
+- Set
+- Sorted Set
+
+Unlike the ingest scenario, write behind has no default behavior for model translation. It is up to the user to create a declarative job, specifying the mapping between Redis key and target database record.
```
Comment (spicy-sauce) [inline content/rdi/architecture.md:153]: Ingest?

Comment (spicy-sauce) [inline content/rdi/architecture.md:153]: Write Behind?

**Review by spicy-sauce — CHANGES_REQUESTED** [review]

File: content/rdi/architecture.md (hunk 2 above)
Comment (yaronp68) [reply content/rdi/architecture.md:12]: we wanted to emphasis it's a product

Comment (yaronp68) [reply content/rdi/architecture.md:12]: done

File: content/rdi/architecture.md (hunk 3 above)
Comment (yaronp68) [reply content/rdi/architecture.md:19]: done

File: content/rdi/architecture.md (hunk 4 above)
Comment (yaronp68) [reply content/rdi/architecture.md:21]: done

File: content/rdi/architecture.md [hunk 16]
```diff
@@ -9,59 +9,89 @@ headerRange: "[2]"
...
+- End to end solution - No need for additional tools and integrations
+- Capture Data Change (CDC) included
+- Covers most popular databases as sources and targets
+- Declarative data mapping and transformations - no custom code needed
+- Data delivery guaranteed (at least once)
+- Zero downtime day 2 operations (reconfigure & upgrade)
+
+Redis Data Integration currently supports two use cases:
+
+- Ingest (cache prefetch).
```
[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:23]: done

File: content/rdi/architecture.md (hunk 5 above)
Comment (yaronp68) [reply content/rdi/architecture.md:24]: done

File: content/rdi/architecture.md [hunk 17]
```diff
@@ -9,59 +9,89 @@ headerRange: "[2]"
...
+- Write Behind.
+
+## Ingest functionality and architecture
+
+You can think of RDI Ingest as a streaming ELT process:
+
+- Data is **E**xtracted from the source database using [Debezium Server](https://debezium.io/) connectors.
 - Data is then **L**oaded into Redis Data Integration, a Redis database instance that keeps the data in [Redis streams](https://redis.io/docs/manual/data-types/streams/) alongside required metadata.
```
[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:31]: done

File: content/rdi/architecture.md [hunk 18]
```diff
@@ -9,59 +9,89 @@ headerRange: "[2]"
...
 - Data is then **T**ransformed using provided [RedisGears recipes](https://developer.redis.com/howtos/redisgears/) and written to the target Redis DB.
 
-Redis Data Integration using Debezium works in 2 modes:
+Redis Data Integration using Debezium Server works in two modes:
```
[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:34]: done

File: content/rdi/architecture.md [hunk 19]
```diff
@@ -97,3 +127,28 @@ Debezium Server should run in warm High Availability topology in one of the foll
...
+
+RDI write behind can track changes to the following Redis types:
+
+- Hash
+- JSON
+- Set
+- Sorted Set
+
+Unlike the ingest scenario, write behind has no default behavior for model translation. It is up to the user to create a declarative job, specifying the mapping between Redis key and target database record.
+The job `keys` and `mapping` sections help make this an easy task.
```
[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:154]: done

File: content/rdi/architecture.md (hunk 15 above)
Comment (yaronp68) [reply content/rdi/architecture.md:153]: they want it all over lowercase

Comment (yaronp68) [reply content/rdi/architecture.md:153]: they want it all over lowercase

File: content/rdi/architecture.md (hunk 8 above)
Comment (yaronp68) [reply content/rdi/architecture.md:133]: it should be lower case all the time

File: content/rdi/architecture.md [hunk 20]
```diff
@@ -9,59 +9,89 @@ headerRange: "[2]"
...
-Currently the only source for Redis Data Integration is the [Debezium Redis Sink Connector](https://debezium.io/documentation/reference/stable/operations/debezium-server.html#_redis_stream).
-Other sources and SDK will be added in later versions.
+Debezium Server has a Redis sink that writes data and state to RDI streams.
+In addition RDI works with [Arcion](arcion.io), a commercial CDC that provides additional set of sources.
 
-## Redis data integration data plane
+#### Redis data integration data plane
 
-### RDI data streams
+##### RDI data streams
```
[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:73]: we can cosider later

File: content/rdi/architecture.md (hunk 7 above)
Comment (yaronp68) [reply content/rdi/architecture.md:69]: done

File: content/rdi/architecture.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:52]: done

Comment (yaronp68) [reply content/rdi/architecture.md:52]: done

File: content/rdi/architecture.md [hunk 21]
```diff
@@ -9,59 +9,89 @@ headerRange: "[2]"
...
+
+- Data is **E**xtracted from the source database using [Debezium Server](https://debezium.io/) connectors.
 - Data is then **L**oaded into Redis Data Integration, a Redis database instance that keeps the data in [Redis streams](https://redis.io/docs/manual/data-types/streams/) alongside required metadata.
 - Data is then **T**ransformed using provided [RedisGears recipes](https://developer.redis.com/howtos/redisgears/) and written to the target Redis DB.
 
-Redis Data Integration using Debezium works in 2 modes:
+Redis Data Integration using Debezium Server works in two modes:
 
 1. Initial sync - Where a snapshot of the entire database or a subset of selected tables is used as a baseline. The entire dataset is streamed to Redis Data Integration and then transformed and written into the target Redis database.
 2. Live updates - Where Debezium captures changes to the data that happen after the baseline snapshot and streams them to Redis Data Integration where they are transformed and written to the target.
```
[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:37]: done

File: content/rdi/architecture.md [hunk 22]
```diff
@@ -9,59 +9,89 @@ headerRange: "[2]"
...
+You can think of RDI Ingest as a streaming ELT process:
+
+- Data is **E**xtracted from the source database using [Debezium Server](https://debezium.io/) connectors.
 - Data is then **L**oaded into Redis Data Integration, a Redis database instance that keeps the data in [Redis streams](https://redis.io/docs/manual/data-types/streams/) alongside required metadata.
 - Data is then **T**ransformed using provided [RedisGears recipes](https://developer.redis.com/howtos/redisgears/) and written to the target Redis DB.
 
-Redis Data Integration using Debezium works in 2 modes:
+Redis Data Integration using Debezium Server works in two modes:
 
 1. Initial sync - Where a snapshot of the entire database or a subset of selected tables is used as a baseline. The entire dataset is streamed to Redis Data Integration and then transformed and written into the target Redis database.
```
[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:36]: done

[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:36]: done

File: content/rdi/architecture.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:142]: done

**Comment by yaronp68** [comment]
@spicy-sauce please note that we are now using caps only for names (e.g. Redis) or at the beginning of the line

File: content/rdi/architecture.md (hunk 20 above)
[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:75]: done

File: content/rdi/architecture.md [hunk 23]
```diff
@@ -97,3 +127,28 @@ Debezium Server should run in warm High Availability topology in one of the foll
...
+RDI engine is installed on a Redis database containing the application data and not on a separate staging Redis database. RDI streams and control plain adds a small footprint of few hundreds of MBs to the Redis database. In the write behind topology, RDI process in parallel on each shard and establishes a single connection from each shard to the downstream database.
+
+### model translation
+
+RDI write behind can track changes to the following Redis types:
+
+- Hash
+- JSON
+- Set
+- Sorted Set
```
[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:154]: done

File: content/rdi/quickstart/ingest-guide.md [hunk 24]
```diff
@@ -15,7 +15,7 @@ This guide will take you through the creation of a write-behind pipeline.
 ## Prerequisites
 
 - Install [RDI CLI]({{<relref "/rdi/installation/install-rdi-cli">}}).
-- An existing Redis Enterprise cluster version >= 6.2.
+- An existing Redis Enterprise cluster version >= 6.2.18
```
[the remark this answers is not in the document]
Comment (spicy-sauce) [reply content/rdi/quickstart/ingest-guide.md:18]: Implemented.

**Review by spicy-sauce — APPROVED** [review]

File: content/rdi/installation/install-redis-gears.md [hunk 25]
```diff
@@ -23,13 +23,13 @@ Download RedisGears based on the Linux distribution of where Redis Enterprise is
 ### Ubuntu 20.04
 
 ```bash
-curl -s https://redismodules.s3.amazonaws.com/redisgears/redisgears.Linux-ubuntu20.04-x86_64.{{<param rdi_redis_gears_version>}}-withdeps.zip -o /tmp/redis-gears.zip
+curl -s --tlsv1.3 https://redismodules.s3.amazonaws.com/redisgears/redisgears.Linux-ubuntu20.04-x86_64.{{<param rdi_redis_gears_current_semantic_version>}}-withdeps.zip -o /tmp/redis-gears.zip
```
Comment (spicy-sauce) [inline content/rdi/installation/install-redis-gears.md:26]: The param name has changed to rdi_redis_gears_version, this won't work.

File: content/rdi/installation/install-redis-gears.md [hunk 26]
```diff
@@ -23,13 +23,13 @@ Download RedisGears based on the Linux distribution of where Redis Enterprise is
...
 ```
 
 ### Ubuntu 18.04
 
 ```bash
-curl -s https://redismodules.s3.amazonaws.com/redisgears/redisgears.Linux-ubuntu18.04-x86_64.{{<param rdi_redis_gears_version>}}-withdeps.zip -o /tmp/redis-gears.zip
+curl -s --tlsv1.3 https://redismodules.s3.amazonaws.com/redisgears/redisgears.Linux-ubuntu18.04-x86_64.{{<param rdi_redis_gears_current_semantic_version>}}-withdeps.zip -o /tmp/redis-gears.zip
```
Comment (spicy-sauce) [inline content/rdi/installation/install-redis-gears.md:32]: The param name has changed to rdi_redis_gears_version, this won't work.

**Review by spicy-sauce — DISMISSED** [review]

File: content/rdi/installation/install-redis-gears.md (hunk 25 above)
Comment (yaronp68) [reply content/rdi/installation/install-redis-gears.md]: done

File: content/rdi/installation/install-redis-gears.md (hunk 26 above)
Comment (yaronp68) [reply content/rdi/installation/install-redis-gears.md]: done

**Review by dwdougherty — APPROVED** [review]

**Review by dwdougherty — APPROVED** [review]

**Pull request merged**